content/400-reference/200-api-reference/300-environment-variables-reference.mdx

```diff
@@ -94,29 +94,49 @@ HTTP_PROXY=http://proxy.example.com
 HTTPS_PROXY=https://proxy.example.com
 ```
 
-## Binary environment variables
+## Engine environment variables
 
-By default all binaries are downloaded when you install Prisma CLI. There are however situations where you may wish to use a custom binary. See [Prisma Engines](../../concepts/components/prisma-engines) for more information.
+### Configuring Query Engine Type
 
-### <inlinecode>PRISMA_BINARIES_MIRROR</inlinecode>
+#### PRISMA_CLI_QUERY_ENGINE_TYPE
 
-`PRISMA_BINARIES_MIRROR` can be used to specify a custom CDN (or server) endpoint to download the engines binaries for the CLI/Client. The default value is `https://binaries.prisma.sh`, where Prisma hosts the binaries.
+`PRISMA_CLI_QUERY_ENGINE_TYPE` is used to [define the query ngine type Prisma CLI downloads and uses](../../concepts/components/prisma-engines/query-engine#configuring-the-query-engine). Defaults to `library`, but can be set to `binary`:
 
-```terminal
-PRISMA_BINARIES_MIRROR=https://my-allowed-url
+```env
+PRISMA_CLI_QUERY_ENGINE_TYPE=binary
 ```
 
-See [Prisma engines](../../concepts/components/prisma-engines#hosting-binaries) for a conceptual overview of how to use this environment variable.
+#### PRISMA_CLIENT_ENGINE_TYPE
 
-### <inlinecode>PRISMA_MIGRATION_ENGINE_BINARY</inlinecode>
+`PRISMA_CLIENT_ENGINE_TYPE` is used to [define the query engine type Prisma Client downloads and uses](../../concepts/components/prisma-engines/query-engine#configuring-the-query-engine). Defaults to `library`, but can be set to `binary`:
 
-`PRISMA_MIGRATION_ENGINE_BINARY` is used to set a custom location for your own migration engine binary.
+```env
+PRISMA_CLIENT_ENGINE_TYPE=binary
+```
+
+Note: You need to generate your Prisma Client after setting this variable for the configuration to take effect during the download of the Prisma Client engines. If you do not do that, running Prisma Client will be missing the appropriate query engine library and you will _need_ to define a path using [`PRISMA_QUERY_ENGINE_LIBRARY`](#prisma-query-engine-library).
```

Comment:
```suggestion
Note: You need to generate your Prisma Client after setting this variable for the configuration to take effect and the libraries to be downloaded. Otherwise, the Prisma Client will be missing the appropriate query engine library and you will _have to_ define their location using [`PRISMA_QUERY_ENGINE_LIBRARY`](#prisma-query-engine-library).
```
The download mention made it possible to think you needed to generate during the download (I know it's kind of twisted :))